code/lib/builder-vite/src/index.ts

```diff
@@ -57,13 +57,25 @@ function iframeMiddleware(options: ExtendedOptions, server: ViteDevServer): Requ
   };
 }
 
+let server: ViteDevServer;
+
+export async function bail(e?: Error): Promise<void> {
+  try {
+    return await server.close();
```

Comment:
Nit: In theory `server` can be `undefined` here - although I can't see if it can in practice right now.

To be defensive I'd change the type to be `let server: ViteDevServer | undefined;` and add optional chaining when using it.

Reply:
Would you want that in order to avoid the console warning?  I think it might be nice to show it, so we know what's happening, but it's also not super actionable, so I could go either way.

Reply:
Hmmm good point, I actually have no strong preference.

Reply:
I considered this but I chose to leave it unguarded so that it would throw and error.